File: C/Signaux/7.2saisit.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <unistd.h>

int main() {
    int n;
    printf("Entrez un entier : ");
    scanf("%d", &n);

    FILE *f = fopen("/tmp/entier.txt", "w");
    fprintf(f, "%d\n", n);
    fclose(f);

    pid_t pid_affiche;
    printf("PID du programme affiche.c : ");
    scanf("%d", &pid_affiche);

    kill(pid_affiche, SIGUSR1); // envoyer signal
    return 0;
}

File: Python/Signaux/7.2saisit.py
import os
import signal

# PID du programme affiche.py
pid_affiche = int(input("Entrez le PID du programme affiche.py : "))

# Saisie de l'entier
n = int(input("Entrez un entier : "))

# Écriture dans un fichier temporaire
with open("/tmp/entier.txt", "w") as f:
    f.write(str(n))

# Envoi du signal SIGUSR1 au programme affiche.py
os.kill(pid_affiche, signal.SIGUSR1)
print(f"Signal envoyé au processus {pid_affiche}")